tests/agent_server/deploy/test_scheduler_agent.py

```diff
@@ -178,6 +214,7 @@ def __init__(
     ):
         super().__init__(environment)
         self.executor_manager = DummyManager()
+        self.scheduler = TestScheduler(self.scheduler.environment, self.scheduler.executor_manager, self.scheduler.client)
```

Comment:
We mix two styles of mocking: subclass Vs overwriting methods. 

I would choose one.